plugins/cpp_metrics/parser/include/cppmetricsparser/cppmetricsparser.h

```diff
@@ -15,12 +15,67 @@
 
 #include <util/parserutil.h>
 #include <util/threadpool.h>
+#include <util/odbtransaction.h>
 
 namespace cc
 {
 namespace parser
 {
-  
+
+/// @brief Constructs an ODB query that you can use to filter only
+/// the database records of the given parameter type whose path
+/// is rooted under any of the specified filter paths.
+/// @tparam TQueryParam The type of database records to query.
+/// This type must represent an ODB view that has access to
+/// (i.e. is also joined with) the File table.
+/// @tparam TIter The iterator type of the filter paths.
+/// @tparam TSentinel The type of the end of the filter paths.
+/// @param begin_ The iterator referring to the first filter path.
+/// @param end_ The sentinel for the end of the filter paths.
+/// @return A query containing the disjunction of filters.
+template<typename TQueryParam, typename TIter, typename TSentinel>
+odb::query<TQueryParam> getFilterPathsQuery(
```

Comment:
This method looks quite independent from C++ metrics and should better be place in `dbutil`.

Reply:
Refactored in 958d418e2861963e68475189fea64b54bd80f9ef.